src/Controls/src/Core/HandlerImpl/Window/Window.Impl.cs

```diff
@@ -454,21 +457,25 @@ void OnModalPushed(Page modalPage)
 			VisualDiagnostics.OnChildAdded(this, modalPage);
 		}
 
-		void OnModalPushing(Page modalPage)
+		internal void OnModalPushing(Page modalPage)
 		{
 			var args = new ModalPushingEventArgs(modalPage);
 			ModalPushing?.Invoke(this, args);
 			Application?.NotifyOfWindowModalEvent(args);
 		}
 
-		void OnPopCanceled()
+		internal void OnPopCanceled()
 		{
 			PopCanceled?.Invoke(this, EventArgs.Empty);
 		}
 
 		void IWindow.Created()
 		{
+			if (IsCreated)
+				throw new InvalidOperationException("Window was already created");
```

Comment:
Didn't we removed something similar to this in other PR ?

Reply:
The original PR used an `#IF DEBUG` so the behavior was different on release vs debug which caused confusing issues. I mainly did that for back port reasons to not change behavior. This is only going into NET8 so I'm fine with changing the behavior a bit in NET8. That's why I added a breaking change label on this PR.